core/arch/arm/plat-rockchip/platform_rk3588.c

```diff
@@ -63,3 +81,61 @@ int platform_secure_ddr_region(int rgn, paddr_t st, size_t sz)
 
 	return 0;
 }
+
+TEE_Result hw_get_random_bytes(void *buf, size_t blen)
+{
+	vaddr_t trng_s_base = (vaddr_t)phys_to_virt_io(TRNG_S_BASE,
+						       TRNG_S_SIZE);
+	uint32_t *rand_buf = (uint32_t *)buf;
+	size_t remaining = blen;
+	uint32_t val;
+
+	if (!trng_s_base)
+		panic("TRNG_S base not mapped");
+
+	/* Ensure TRNG is seeded and ready */
+	val = io_read32(trng_s_base + TRNG_S_STAT);
+	if (!(val & (1 << 9))) {
+		/* TRNG not seeded, issue SEED command */
+		io_write32(trng_s_base + TRNG_S_CTRL, CMD_SEED);
+
+		/* Wait for SEED_DONE flag in ISTAT register */
+		do {
+			val = io_read32(trng_s_base + TRNG_S_ISTAT);
+		} while (!(val & (1 << 1)));
+
+		/* SEED_DONE flag set, clear SEED_DONE */
+		io_write32(trng_s_base + TRNG_S_ISTAT, (1 << 1));
+	}
+
+	/* Set RNG length to 128 bits and disable interrupts */
+	io_write32(trng_s_base + TRNG_S_MODE, LEN_128BIT);
+	io_write32(trng_s_base + TRNG_S_IE, 0);
```

Comment:
What interrupts? From the TRNG? If so why is it done here in the middle of the function and not earlier?